apps/arweave/src/ar_verify_chunks.erl

```diff
@@ -148,10 +150,56 @@ verify_proof(Metadata, State) ->
 				read_data_path_error, AbsoluteOffset, ChunkSize, [{reason, Error}], State)
 	end.
 
-verify_chunk_storage(PaddedOffset, _ChunkSize, {End, Start}, State) 
+verify_packing(Metadata, State) ->
+	#state{packing=Packing, store_id=StoreID} = State,
+	{AbsoluteOffset, ChunkDataKey, TXRoot, _DataRoot, TXPath,
+			_TXRelativeOffset, ChunkSize} = Metadata,
+	PaddedOffset = ar_block:get_chunk_padded_offset(AbsoluteOffset),
+	StoredPackingCheck = ar_sync_record:is_recorded(AbsoluteOffset, ar_data_sync, StoreID),
+	ExpectedPacking =
+		case ar_chunk_storage:is_storage_supported(PaddedOffset, ChunkSize, Packing) of
+			true ->
+				Packing;
+			false ->
+				unpacked
+		end,
+	case {StoredPackingCheck, ExpectedPacking} of
+		{{true, ExpectedPacking}, _} ->
+			ok;
+		{{true, unpacked_padded}, {replica_2_9, _}} ->
+			%% The module is in the process of entropy generation but has already
+			%% synced some chunks.
+			ok;
+		{{true, StoredPacking}, _} ->
+			?LOG_WARNING([{event, verify_chunk_storage_unexpected_packing},
+				{expected_packing, ar_storage_module:packing_label(Packing)},
+				{stored_packing, ar_storage_module:packing_label(StoredPacking)},
+				{offset, AbsoluteOffset}]),
+			ar_data_sync:invalidate_bad_data_record(AbsoluteOffset, ChunkSize, StoreID,
+					unexpected_packing);
+		{Reply, _} ->
+			?LOG_WARNING([{event, verify_chunk_storage_missing_packing_info},
+				{expected_packing, ar_storage_module:packing_label(Packing)},
+				{packing_reply, io_lib:format("~p", [Reply])},
+				{offset, AbsoluteOffset}]),
+			ar_data_sync:invalidate_bad_data_record(AbsoluteOffset, ChunkSize, StoreID,
+					missing_packing_info)
+	end,
+	State.
+
+verify_chunk_storage(AbsoluteOffset, PaddedOffset, ChunkSize, {End, Start}, State)
 		when PaddedOffset - ?DATA_CHUNK_SIZE >= Start andalso PaddedOffset =< End ->
+	#state{store_id = StoreID} = State,
+	case ar_chunk_storage:read_offset(PaddedOffset, StoreID) of
```

Comment:
Do we need to make sure we only run this test on `is_storage_supported` chunks? or does that check happen earlier?

Reply:
The `PaddedOffset - ?DATA_CHUNK_SIZE >= Start andalso PaddedOffset =< End` check indicates the chunk is recorded in ar_chunk_storage.